src/main/java/io/lettuce/core/RedisJsonCommandBuilder.java

```diff
@@ -106,13 +106,17 @@ Command<K, V, List<JsonValue>> jsonArrpop(K key, JsonPath jsonPath, int index) {
 
         CommandArgs<K, V> args = new CommandArgs<>(codec).addKey(key);
 
-        if (jsonPath != null && !jsonPath.isRootPath()) {
-            // OPTIONAL as per API
-            args.add(jsonPath.toString());
-
-            if (index != -1) {
-                // OPTIONAL as per API
-                args.add(index);
+        if (jsonPath != null) {
+            if (!jsonPath.isRootPath()) {
+                args.add(jsonPath.toString());
+                if (index != -1) {
+                    args.add(index);
+                }
+            } else {
+                if (index != -1) {
+                    args.add(jsonPath.toString());
+                    args.add(index);
+                }
```

Comment:
Hey, @thachlp , thanks for the contribution!

What do you think about his simpler approach?
```suggestion
        if (jsonPath != null) {
            // OPTIONAL as per API
            args.add(jsonPath.toString());

            if (index != -1) {
                // OPTIONAL as per API
                args.add(index);
```

Reply:
@tishun Great, it is simpler.
But this test case will fail:
```
shouldCorrectlyConstructJsonArrpopRootPathNoIndex
```
Because current, we ignore the root path if the index = -1
So should we change the logic?

Reply:
@tishun  I updated the logic and the test 🙇

Reply:
Oh, oh, no, you are quite right, we should be able to do that too.

Reply:
Sorry about that